includes/data-events/listeners.php

```diff
@@ -253,20 +253,18 @@ function ( $order_id, $order ) {
 		if ( ! $product_id ) {
 			return;
 		}
-		$recurrence      = get_post_meta( $product_id, '_subscription_period', true );
-		$is_renewal      = function_exists( 'wcs_order_contains_renewal' ) && wcs_order_contains_renewal( $order );
-		$subscription_id = null;
-		if ( function_exists( 'wcs_get_subscriptions_for_renewal_order' ) ) {
-			$subscriptions   = array_values( wcs_get_subscriptions_for_renewal_order( $order ) );
-			$subscription_id = is_array( $subscriptions ) && ! empty( $subscriptions ) && is_a( $subscriptions[0], 'WC_Subscription' ) ? $subscriptions[0]->get_id() : null;
-		}
+		$recurrence       = \get_post_meta( $product_id, '_subscription_period', 'once' );
```

Comment:
The 3rd argument of post meta is not the default value. It's a bool for whether to return a single value or not.

Reply:
Incidentally fixed by removing `recurrence` from the payload in 74b8fb80 (see https://github.com/Automattic/newspack-plugin/pull/2821#discussion_r1450851895)

Reply:
It should also be reverted for the `donation_new` listener.

Reply:
Ugh, sorry 6570674e